clang/include/clang/Sema/ParsedAttr.h

```diff
@@ -655,6 +655,28 @@ class ParsedAttr final
     }
   }
 
+  /// If this is an OpenCL address space attribute returns its SYCL
+  /// representation in LangAS, otherwise returns default address space.
+  LangAS asSYCLLangAS() const {
+    switch (getKind()) {
+    case ParsedAttr::AT_OpenCLConstantAddressSpace:
+      return LangAS::opencl_constant;
+    case ParsedAttr::AT_OpenCLGlobalAddressSpace:
+      return LangAS::sycl_global;
+    case ParsedAttr::AT_OpenCLGlobalDeviceAddressSpace:
+      return LangAS::sycl_global_device;
+    case ParsedAttr::AT_OpenCLGlobalHostAddressSpace:
+      return LangAS::sycl_global_host;
+    case ParsedAttr::AT_OpenCLLocalAddressSpace:
+      return LangAS::sycl_local;
+    case ParsedAttr::AT_OpenCLPrivateAddressSpace:
+      return LangAS::sycl_private;
+    case ParsedAttr::AT_OpenCLGenericAddressSpace:
+    default:
+      return LangAS::Default;
```

Comment:
> It looks odd to radically change to `Default`, this could potentially have side effects for the host code (like overload redefinition).

Is this really a **change**? I think `Default` address space is used by default, so I don't expect a "radical change".
NOTE: OpenCL is using the same mapping.

> 
> I would suggest to keep unchanged the address space if not in the handled cases.

I'm not sure what do you mean. This function maps `ParsedAttr` to `LangAs` enum. According to my understanding using `Default` address space by default achieves the described behavior.

Reply:
> I'm not sure what do you mean. This function maps ParsedAttr to LangAs enum.

Oups, I completely miss-read this. For some reason, I got stuck into "this function does LangAs to LangAs"...

So, the function LGTM :)